axelor-budget/src/main/java/com/axelor/apps/budget/web/GlobalBudgetViewController.java

```diff
@@ -0,0 +1,156 @@
+package com.axelor.apps.budget.web;
+
+import com.axelor.apps.account.db.repo.InvoiceRepository;
+import com.axelor.apps.account.db.repo.MoveRepository;
+import com.axelor.apps.budget.db.Budget;
+import com.axelor.apps.budget.db.BudgetDistribution;
+import com.axelor.apps.budget.db.GlobalBudget;
+import com.axelor.apps.budget.db.repo.GlobalBudgetRepository;
+import com.axelor.apps.budget.service.GlobalBudgetService;
+import com.axelor.apps.purchase.db.repo.PurchaseOrderRepository;
+import com.axelor.apps.sale.db.repo.SaleOrderRepository;
+import com.axelor.i18n.I18n;
+import com.axelor.inject.Beans;
+import com.axelor.meta.schema.actions.ActionView;
+import com.axelor.rpc.ActionRequest;
+import com.axelor.rpc.ActionResponse;
+import com.google.common.base.Joiner;
+import java.util.List;
+
+public class GlobalBudgetViewController {
+
+  public void viewGlobalPurchaseOrderLine(ActionRequest request, ActionResponse response) {
+    GlobalBudget globalBudget = request.getContext().asType(GlobalBudget.class);
+    List<Long> budgetIdList = Beans.get(GlobalBudgetService.class).getAllBudgetIds(globalBudget);
+
+    String domain =
+        String.format(
+            "self.purchaseOrderLine.purchaseOrder.statusSelect in (%d,%d,%d) and self.budget.id IN (%s)",
+            PurchaseOrderRepository.STATUS_REQUESTED,
+            PurchaseOrderRepository.STATUS_VALIDATED,
+            PurchaseOrderRepository.STATUS_FINISHED,
+            Joiner.on(",").join(budgetIdList));
```

Comment:
a nice way to improve this is to set `budgetIdList` as context, and use it as a parameter in the domain